File: c/readability/readability.c
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int letras = 0, palavras = 1, frases = 0;
    char texto[1000];

    // Obter o texto do usuário
    printf("Text: ");
    fgets(texto, sizeof(texto), stdin);

    // Iterar sobre cada caractere do texto
    for (int i = 0, n = strlen(texto); i < n; i++)
    {
        if (isalpha(texto[i]))
        {
            letras++;
        }
        else if (texto[i] == ' ')
        {
            palavras++;
        }
        else if (texto[i] == '.' || texto[i] == '!' || texto[i] == '?')
        {
            frases++;
        }
    }

    // Calcular L e S
    float L = (float) letras / palavras * 100;
    float S = (float) frases / palavras * 100;

    // Calcular o índice Coleman-Liau
    float indice = 0.0588 * L - 0.296 * S - 15.8;

    // Arredondar o índice para o inteiro mais próximo
    int grau = round(indice);

    // Exibir o grau correspondente
    if (grau < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grau >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grau);
    }

    return 0;
}

File: python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Solicita ao usuário um texto
    text = get_string("Text: ")

    # Inicializa contadores
    letters = 0
    words = 0
    sentences = 0

    # Percorre cada caractere no texto
    for char in text:
        if char.isalpha():  # Conta letras
            letters += 1
        elif char == ' ':  # Conta palavras
            words += 1
        elif char in ['.', '!', '?']:  # Conta sentenças
            sentences += 1

    # Adiciona 1 ao contador de palavras para contar a última palavra
    words += 1

    # Calcula L e S
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Calcula o índice de Coleman-Liau
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Determina a saída conforme o índice
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


if __name__ == "__main__":
    main()
